File: tests/test_tts_hash.mjs
const testCases = [
    ["Hello World", "vi-VN-Standard-A", "vi-VN"],
    ["Xin chào, hôm nay trời đẹp quá!", "vi-VN-Wavenet-B", "vi-VN"],
    ["Tiếng Việt có dấu phức tạp", "en-US-Standard-C", "en-US"],
];

async function generateHash(text, voice, lang) {
    const rawString = `${text}|${voice}|${lang}`;
    const encoder = new TextEncoder();
    const data = encoder.encode(rawString);
    const hashBuffer = await globalThis.crypto.subtle.digest('SHA-256', data);
    const hashArray = Array.from(new Uint8Array(hashBuffer));
    return hashArray.map(b => b.toString(16).padStart(2, '0')).join('').substring(0, 16);
}

(async () => {
    console.log("--- JS (Node) Hash Output ---");
    for (const [text, voice, lang] of testCases) {
        const h = await generateHash(text, voice, lang);
        console.log(`${h}  <- '${text}'`);
    }
})();

File: tests/test_tts_hash.py
import hashlib

def generate_hash(text, voice, lang):
    raw_data = f"{text}|{voice}|{lang}"
    return hashlib.sha256(raw_data.encode('utf-8')).hexdigest()[:16]

test_cases = [
    ("Hello World", "vi-VN-Standard-A", "vi-VN"),
    ("Xin chào, hôm nay trời đẹp quá!", "vi-VN-Wavenet-B", "vi-VN"),
    ("Tiếng Việt có dấu phức tạp", "en-US-Standard-C", "en-US"),
]

print("--- Python Hash Output ---")
for text, voice, lang in test_cases:
    print(f"{generate_hash(text, voice, lang)}  <- '{text}'")
